clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -51,6 +52,11 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
         this.topicPartition = topicPartition;
     }
 
+    @Deprecated
+    public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
+        this(topicPartition, baseOffset, relativeOffset, Record.NO_TIMESTAMP, -1, -1, -1);
```

Comment:
I can add them if you like. Since they would only be used in the case of a deprecated method (or tests that don't care) I didn't add them thinking the would just be removed when the constructor is removed.